lib/virt_autotest/utils.pm

```diff
@@ -943,5 +942,44 @@ sub download_vm_import_disks {
     record_info("All imported disk download is done.");
 }
 
+sub enable_nm_debug {
+    # Enable Network Manager Debug Log Level
+    assert_script_run("nmcli general logging level DEBUG domains ALL", 60);
+    record_info("Enable Network Manager in Debug Level successfully for automation test.");
+}
+
+sub check_activate_network_interface {
+    # Check with activate network interface as required
+    my $target = "openqa.suse.de";
+    my $nic = "br0";
+    assert_script_run("ping -I $nic -c 3 $target", 60);
+    assert_script_run("nmcli con show $nic", 60);
+    save_screenshot;
+    record_info("Activate Network Interface check successfully for automation test.");
+}
+
+sub set_host_bridge_interface_with_nm {
+    # Setup Host Bridge Network Interface with nmcli(NetworkManager) as needed
+    my $_host_bridge_cfg = "/etc/NetworkManager/system-connections/br0.nmconnection";
+
+    # Change the NetworkManager log-level as DEBUG at runtime
+    enable_nm_debug;
+
+    if (script_run("[[ -f $_host_bridge_cfg ]]") != 0) {
```

Comment:
We can only save br0 setup if br0 is working well, right? Checking nmconnection file will hide some not-working scenarios?

Reply:
Emm,  good point.  just only check with br0.nmconnection would be not enough as expected.  we should double check with br0-slave.nmconnection and br0.nmconnection files together.

Reply:
Sorry, please ignore comment on this, `check_activate_network_interface` is called anyway , so should be fine.